api/openapi/model-registry.yaml

```diff
@@ -1736,6 +1860,166 @@ components:
             servingEnvironmentId:
               description: ID of the parent `ServingEnvironment` for this `InferenceService` entity.
               type: string
+    CatalogSourceList:
+      description: List of CatalogSource entities.
+      allOf:
+        - type: object
+          properties:
+            items:
+              description: Array of `CatalogSource` entities.
+              type: array
+              items:
+                $ref: "#/components/schemas/CatalogSource"
+        - $ref: "#/components/schemas/BaseResourceList"
+    CatalogSource:
+      description: A catalog source. A catalog source has CatalogModel children.
+      required:
+        - name
+      type: object
+      properties:
+        name:
+          description: The name of the catalog source.
+          type: string
+    CatalogModelList:
+      description: List of CatalogModel entities.
+      allOf:
+        - type: object
+          properties:
+            items:
+              description: Array of `CatalogModel` entities.
+              type: array
+              items:
+                $ref: "#/components/schemas/CatalogModel"
+          required:
+            - items
+        - $ref: "#/components/schemas/BaseResourceList"
+    CatalogModel:
```

Comment:
> can we create a PR with the base types

I agree, we can use `CatalModel` as base type if it has all the common shared properties.
If we prefer to have a different base entity that's fine too but we should really limit the hierarchy of model metadata to avoid possible incompatibilities/gaps when we convert one "model type" (i.e. catalog model) in another "model type" (i.e. registered model)